src/dlgTriggerEditor.cpp

```diff
@@ -3769,111 +4646,91 @@ void dlgTriggerEditor::slot_trigger_selected(QTreeWidgetItem* pItem)
+                pItem->pushButton_bgColor->hide();
+                pItem->lineEdit_pattern->show();
                 break;
             case REGEX_EXACT_MATCH:
                 palette.setColor(QPalette::Text, QColor(0, 195, 0));
                 pBox->setCurrentIndex(3);
-                pItem->fgB->hide();
-                pItem->bgB->hide();
-                pItem->lineEdit->show();
+                pItem->pushButton_fgColor->hide();
+                pItem->pushButton_bgColor->hide();
+                pItem->lineEdit_pattern->show();
                 break;
             case REGEX_LUA_CODE:
                 palette.setColor(QPalette::Text, QColor(0, 155, 155));
                 pBox->setCurrentIndex(4);
-                pItem->fgB->hide();
-                pItem->bgB->hide();
-                pItem->lineEdit->show();
+                pItem->pushButton_fgColor->hide();
+                pItem->pushButton_bgColor->hide();
+                pItem->lineEdit_pattern->show();
                 break;
             case REGEX_LINE_SPACER:
                 palette.setColor(QPalette::Text, QColor(137, 0, 205));
                 pBox->setCurrentIndex(5);
-                pItem->fgB->hide();
-                pItem->bgB->hide();
-                pItem->lineEdit->show();
+                pItem->pushButton_fgColor->hide();
+                pItem->pushButton_bgColor->hide();
+                pItem->lineEdit_pattern->show();
                 break;
             case REGEX_COLOR_PATTERN:
                 palette.setColor(QPalette::Text, QColor(100, 100, 100));
                 pBox->setCurrentIndex(6);
-                pItem->fgB->show();
-                pItem->bgB->show();
-                pItem->lineEdit->hide();
-                QPalette fgPal;
-                QPalette bgPal;
+                pItem->pushButton_fgColor->show();
+                pItem->pushButton_bgColor->show();
+                pItem->lineEdit_pattern->hide();
                 if (!pT->mColorPatternList[i]) {
                     break;
                 }
-                auto fgC = QColor(pT->mColorPatternList[i]->fgR, pT->mColorPatternList[i]->fgG, pT->mColorPatternList[i]->fgB);
-                fgPal.setColor(QPalette::Button, fgC);
-                QString fgstyleSheet = QString("QPushButton{background-color:") + fgC.name() + QString(";}");
-                auto bgC = QColor(pT->mColorPatternList[i]->bgR, pT->mColorPatternList[i]->bgG, pT->mColorPatternList[i]->bgB);
-                bgPal.setColor(QPalette::Button, bgC);
-                QString bgstyleSheet = QString("QPushButton{background-color:") + bgC.name() + QString(";}");
-                pItem->fgB->setStyleSheet(fgstyleSheet);
-                pItem->bgB->setStyleSheet(bgstyleSheet);
+                pItem->pushButton_fgColor->setStyleSheet(QStringLiteral("QPushButton{background-color: %1;}").arg(QColor(pT->mColorPatternList[i]->fgR, pT->mColorPatternList[i]->fgG, pT->mColorPatternList[i]->fgB).name()));
```

Comment:
We should not be mixing `QStyle` and `QPallet` on the same widget - that way madness lies {and the Qt Docs say ___don't___} - it is just as well that the pallet stuff isn't used anyhow! 😆